autofaiss/indices/distributed.py

```diff
@@ -479,11 +480,17 @@ def _create_and_train_index_from_embedding_dir() -> TrainedIndex:
 
     partition = extract_partition_name_from_path(embedding_root_dir)
 
-    # Train index
-    rdd = ss.sparkContext.parallelize([embedding_root_dir], 1)
-    trained_index_path, trained_index_key, _, = rdd.map(
-        lambda _: _create_and_train_index_from_embedding_dir()
-    ).collect()[0]
+    if not index_path:
+        # Train index
+        rdd = ss.sparkContext.parallelize([embedding_root_dir], 1)
+        trained_index_path, trained_index_key, _, = rdd.map(
+            lambda _: _create_and_train_index_from_embedding_dir()
+        ).collect()[0]
+    else:
+        # index key of the input index must be provided
+        assert index_key
```

Comment:
you could add the comment in the assert 
assert index_key, "index key of the input index must be provided because you provided an index_path"